pkg/controller/endpointslicemirroring/reconciler.go

```diff
@@ -197,14 +199,9 @@ func (r *reconciler) reconcileByPortMapping(
 		// if >0 existing slices, mark all but 1 for deletion.
 		slices.toDelete = existingSlices[1:]
 
-		// generated slices must mirror all endpoints annotations but EndpointsLastChangeTriggerTime
-		compareAnnotations := cloneAndRemoveKeys(endpoints.Annotations, corev1.EndpointsLastChangeTriggerTime)
-		compareLabels := cloneAndRemoveKeys(existingSlices[0].Labels, discovery.LabelManagedBy, discovery.LabelServiceName)
-		// Return early if first slice matches desired endpoints, labels and annotations
+		// Return early if first slice matches desired endpoints.
 		totals = totalChanges(existingSlices[0], desiredSet)
-		if totals.added == 0 && totals.updated == 0 && totals.removed == 0 &&
-			apiequality.Semantic.DeepEqual(endpoints.Labels, compareLabels) &&
-			apiequality.Semantic.DeepEqual(compareAnnotations, existingSlices[0].Annotations) {
+		if totals.added == 0 && totals.updated == 0 && totals.removed == 0 {
```

Comment:
you are modifying the behaviour, the goal is to consolidate the code from both controllers into one place, the end results should not modify any behaviour
maybe my comment about using previous PR as guidance was misleading?

Reply:
rebased.